test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/CcmBridge.java

```diff
@@ -354,6 +358,10 @@ private Builder() {
       }
       // disable auto_snapshot by default to reduce disk usage when destroying schema.
       withCassandraConfiguration("auto_snapshot", "false");
+      // mark the ccm temp directories for deletion when the JVM exits
+      if (this.configDirectory != null) {
+        this.configDirectory.toFile().deleteOnExit();
```

Comment:
Are we sure this directory will be empty? If not, I suggest doing the following instead:

```
Runtime.getRuntime()
    .addShutdownHook(
        new Thread(
            () -> {
              try (Stream<Path> walk = Files.walk(configDirectory)) {
                walk.sorted(Comparator.reverseOrder())
                    .map(Path::toFile)
                    .forEach(File::delete);
              } catch (IOException e) {
                logger.error("Could not walk " + configDirectory, e);
              }
            }));
```

Reply:
I'm pretty sure the directory will be empty. It seems that when the tests finish, they execute a `ccm remove` that cleans up everything in the directory, just not the directory itself. I did try to setup a shutdown hook to clean up the directory, but it seems to cause issues where the ccm process hangs at the completion of the test.